### ISSUE TITLE
Octane memory building up

### ISSUE THREAD

**Issue by mennorenkens** [spatie/laravel-medialibrary#2495]
Installed [Laravel Octane](https://laravel.com/docs/8.x/octane) on our project and after some debugging it seemed as if interacting with the MediaLibrary builds up memory on each request. I reproduced it in a fresh Laravel installation:
- Create a new Laravel project
- Install Laravel Octane
- Install MediaLibrary, publish the migration file, and add the interface and trait to the User.php model
- Connect a DB and migrate
- Add some users in the database

Then  implement the `registerMediaCollections(`) function:

```php
public function registerMediaCollections(): void
    {
        $this->addMediaCollection('cover')
            ->singleFile()
            ->useDisk('media');
    }
```

And add the following to e.g. `routes/api.php`:

```php
Route::get('test', function () {
    return User::with('media')->get()->map(fn ($user) => $user->getFirstMediaUrl('cover'));
    // alternative without side effect: return User::with('media')->get()->map(fn ($user) => $user->name);
});
```

Now when hitting this route the mb/request slowly builds up when serving the app via Laravel Octane, whereas it does not when the commented code is used instead. I'm using all latest versions: Laravel 8.48.2, PHP 8.0.2, MySQL 8.0 and MediaLibrary 9.6.4.

When interacting with MediaLibrary:
<img width="638" alt="Screenshot 2021-06-26 at 17 32 56" src="https://user-images.githubusercontent.com/18008421/123518074-9fdd6780-d6a4-11eb-8777-dab97fa6c719.png">

Without:
<img width="638" alt="Screenshot 2021-06-26 at 17 33 11" src="https://user-images.githubusercontent.com/18008421/123518080-a23fc180-d6a4-11eb-981b-37a793cad1c3.png">

**Comment by freekmurze** [comment]
Could you create a PR with the proposed fix? Thanks!

**Comment by mennorenkens** [comment]
> Could you create a PR with the proposed fix? Thanks!

@freekmurze tried to dive into the code before creating this issue but couldn't figure out what the cause was. Do you have any clue or could you give me maybe a direction to look at?
In any case, I'll have a look somewhere the coming period and see if I can fix it :).

**Comment by mennorenkens** [comment]
@freekmurze turned out to be caused by the two closures in the constructor of the MediaCollection.php file: 
https://github.com/spatie/laravel-medialibrary/blob/232f347ef7ea9f373d00fbc6280e6fb9489330ec/src/MediaCollections/MediaCollection.php#L41-L44. 
It's not a bug, but without manually setting them to null or without manually calling the garbage collector, they will stay in memory: https://bugs.php.net/bug.php?id=75914

Static anonymous functions could solve it, as they have no $this reference, but some people using the package might need it (https://www.designcise.com/web/tutorial/what-are-static-anonymous-functions-in-php).

For now I just lowered the garbage collection threshold in `config/octane.php`. Tried it initially but didn't notice that the CollectGarbage listener in the same config file is commented by default. At least I learned something 🙃!
https://github.com/laravel/octane/blob/7c56189d3db7c99093ef3af14b33720e858918c6/config/octane.php#L105-L106
